datalad/runner/runner.py

```diff
@@ -191,7 +196,7 @@ def run(self,
         if issubclass(protocol, GeneratorMixIn):
             return results_or_iterator
         else:
-            results = results_or_iterator
+            results = cast(dict, results_or_iterator)
```

Comment:
For myself a note that according to [docs](https://docs.python.org/3/library/typing.html#typing.cast) - "This returns the value unchanged." so should have no effect at runtime. good.